server/models/user.js

```diff
@@ -0,0 +1,34 @@
+const { Schema, model } = require('mongoose');
+const bcrypt = require('bcrypt');
+
+// creating userSchema
+const userSchema = new Schema({
+  username: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+});
+
+// hash user password
+userSchema.pre('save', async function (next) {
+  if (this.isNew || this.isModified('password')) {
+    const saltRounds = 10;
+    this.password = await bcrypt.hash(this.password, saltRounds);
+  }
+
+  next();
+});
+
+// custom method to compare and validate password for logging in
+userSchema.methods.isCorrectPassword = async function (password) {
+  return bcrypt.compare(password, this.password);
+};
+
+const user = model('user', userSchema);
+
+module.exports = user;
```

Comment:
user variable and string needs to be capital

Reply:
I am on it